pennylane/_qubit_device.py

```diff
@@ -1436,6 +1436,53 @@ def var(self, observable, shot_range=None, bin_size=None):
         # TODO: do we need to squeeze here? Maybe remove with new return types
         return np.squeeze(np.var(samples, axis=axis))
 
+    def _samples_to_counts(self, samples, obs, num_wires):
+        """Groups the samples into a dictionary showing number of occurences for
+        each possible outcome.
+
+        Args:
+            samples: samples in an array of dimension ``(shots,len(wires))``
+            obs (Observable): the observable sampled
+            num_wires (int): number of wires the sampled observable was performed on
+
+        Returns:
+            dict: dictionary with format {'outcome': num_occurences}, including all
+                outcomes for the sampled observable
+
+        **Example**
+
+            >>> samples
+            tensor([[0, 0],
+                    [0, 0],
+                    [1, 0]], requires_grad=True)
+            >>> self._samples_to_counts(samples, obs, num_wires)
+            {'00': 2, '01': 0, '10': 1, '11': 0}
+        """
+
+        if isinstance(obs, MeasurementProcess):
+            outcomes = self.generate_basis_states(num_wires)
+
+            # ToDo: converting to str means code does not work with JAX
+            # convert samples and outcomes from arrays to str for dict keys
```

Comment:
No, as far as I know it should be fine. I just added this at one point because I'm not very familiar with JAX and wasn't sure what the comments in the code about JAX compatibility were referring to, so I wanted to remember to check - and then forgot to remove it.